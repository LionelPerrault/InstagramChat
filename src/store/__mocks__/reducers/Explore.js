export default {
  suggested: [
    {
      id: 7,
      username: 'ghalib',
      firstname: 'Mirza',
      surname: 'Ghalib',
      mutualUsersCount: 1,
    },
    {
      id: 8,
      username: 'coldplay',
      firstname: 'cold',
      surname: 'play',
      mutualUsersCount: 0,
    },
  ],
  users: [
    {
      id: 7,
      username: 'ghalib',
      firstname: 'Mirza',
      surname: 'Ghalib',
      mutualUsersCount: 1,
      followers_count: 2,
    },
    {
      id: 8,
      username: 'coldplay',
      firstname: 'cold',
      surname: 'play',
      mutualUsersCount: 0,
      followers_count: 4,
    },
  ],
  groups: [
    {
      group_id: 12,
      name: 'nmnmnmnm',
      created: '1480114098767',
      admin: 7,
      joined: false,
      membersCount: 3,
      mutualMembersCount: 3,
    },
    {
      group_id: 11,
      name: 'a groupss',
      created: '1480114098767',
      admin: 24,
      joined: false,
      membersCount: 4,
      mutualMembersCount: 0,
    },
  ],
  photos: [
    {
      imgsrc: 'instagram_1516524098767.jpg',
      filter: 'filter-normal',
      user: 14,
      username: 'kinkade',
      firstname: 'Thomas',
      surname: 'Kinkade',
      post_id: 28,
      post_time: '1480114098767',
    },
  ],
}
