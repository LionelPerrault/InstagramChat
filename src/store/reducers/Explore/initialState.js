export default {
  users: [],
  photos: [],
  groups: [],
  suggested: [],
}
