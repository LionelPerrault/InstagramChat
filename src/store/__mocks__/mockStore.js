import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'

// mock-reducers
import User from './reducers/User'
import Notification from './reducers/Notification'
import Group from './reducers/Group'
import Follow from './reducers/Follow'
import Explore from './reducers/Explore'
import Post from './reducers/Post'
import Hashtag from './reducers/Hashtag'
import Setting from './reducers/Setting'
import Message from './reducers/Message'

const initialState = {
  User,
  Notification,
  Group,
  Follow,
  Explore,
  Post,
  Hashtag,
  Setting,
  Message,
}

const middlewares = [thunk]
const createStore = configureStore(middlewares)
const mockStore = createStore(initialState)

export default mockStore
