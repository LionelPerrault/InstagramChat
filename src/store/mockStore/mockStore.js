import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import logger from 'redux-logger'
import promise from 'redux-promise-middleware'

// mock-reducers
import User from './mock-reducers/User'
import Notification from './mock-reducers/Notification'
import Group from './mock-reducers/Group'
import Follow from './mock-reducers/Follow'
import Explore from './mock-reducers/Explore'

const initialState = {
  User,
  Notification,
  Group,
  Follow,
  Explore
}

const middlewares = [
  promise(), thunk, logger
]
const createStore = configureStore(middlewares)
const mockStore = createStore(initialState)

export default mockStore
