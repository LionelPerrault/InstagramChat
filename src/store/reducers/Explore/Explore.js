/* eslint indent:0 */
/* eslint no-unreachable:0 */

import initialState from './initialState'

export default (state = initialState, action) => {
  let py = action.payload

  switch (action.type) {
    case 'GET_USERS_TO_EXPLORE':
      return { ...state, users: py }
      break

    case 'GET_PHOTOS_TO_EXPLORE':
      return { ...state, photos: py }
      break

    case 'GET_GROUPS_TO_EXPLORE':
      return { ...state, groups: py }
      break

    case 'GET_SUGGESTED_USERS':
      return { ...state, suggested: py }
      break
  }

  return state
}
