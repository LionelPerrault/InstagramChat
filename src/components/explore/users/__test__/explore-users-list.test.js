import React from 'react'
import { Provider } from 'react-redux'
import mockStore from '../../../../store/__mocks__/mockStore'
import ExploreUsersList, { PureExploreUsersList } from '../explore-users-list'
import Explore from '../../../../store/__mocks__/reducers/Explore'
import { create } from 'react-test-renderer'
import { shallow } from 'enzyme'

describe('ExploreUsersList Component', () => {
  const comp = (index = 0) => (
    <Provider store={mockStore}>
      <ExploreUsersList {...Explore.users[index]} />
    </Provider>
  )

  it('should match snapshot', () => {
    const tree = create(comp()).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should match snapshot when mutualUsersCount = 0', () => {
    const tree = create(comp(1)).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should toggle <MonSticky/> when hovered over the top-parent div', () => {
    const wrapper = shallow(<PureExploreUsersList {...Explore.users[0]} />)

    // should show
    wrapper.find('.m_on').simulate('mouseOver')
    expect(wrapper.find('MonSticky').prop('show')).toEqual(true)

    // should hide
    wrapper.find('.m_on').simulate('mouseOut')
    expect(wrapper.find('MonSticky').prop('show')).toEqual(false)
  })
})
