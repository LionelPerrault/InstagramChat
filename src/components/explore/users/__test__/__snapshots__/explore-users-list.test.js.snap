// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExploreUsersList Component should match snapshot 1`] = `
<div
  className="m_on followers_m_on"
  onMouseOut={[Function]}
  onMouseOver={[Function]}
>
  <div
    className="m_top"
  >
    <img
      src="/users/7/avatar.jpg"
    />
    <div
      className="m_top_right"
    >
      <Link
        to="/profile/ghalib"
      >
        ghalib
      </Link>
      <span>
        1 mutual follower
      </span>
    </div>
  </div>
  <div
    className="m_bottom"
  >
    <a
      className="pri_btn follow"
      href="#"
      onClick={[Function]}
    >
      Follow
    </a>
  </div>
</div>
`;

exports[`ExploreUsersList Component should match snapshot when mutualUsersCount = 0 1`] = `
<div
  className="m_on followers_m_on"
  onMouseOut={[Function]}
  onMouseOver={[Function]}
>
  <div
    className="m_top"
  >
    <img
      src="/users/8/avatar.jpg"
    />
    <div
      className="m_top_right"
    >
      <Link
        to="/profile/coldplay"
      >
        coldplay
      </Link>
      <span>
        cold play
      </span>
    </div>
  </div>
  <div
    className="m_bottom"
  >
    <a
      className="pri_btn follow"
      href="#"
      onClick={[Function]}
    >
      Follow
    </a>
  </div>
</div>
`;
